packages/qwik/src/core/use/use-error-boundary.ts

```diff
@@ -1,16 +1,11 @@
-import { qrl } from '../qrl/qrl';
 import { type ErrorBoundaryStore, ERROR_CONTEXT } from '../render/error-handling';
-import { useContextProvider } from './use-context';
-import { useOn } from './use-on';
+import { useContext, useContextProvider } from './use-context';
 import { useStore } from './use-store.public';
 
 /** @public */
-export const useErrorBoundary = (): Readonly<ErrorBoundaryStore> => {
-  const store: ErrorBoundaryStore = useStore({
-    error: undefined,
-  });
-  useOn('error-boundary', qrl('/runtime', 'error', [store]));
-  useContextProvider(ERROR_CONTEXT, store);
+export const useErrorBoundary = () => {
+  const error = useStore<ErrorBoundaryStore>({ error: undefined });
+  useContextProvider(ERROR_CONTEXT, error);
 
-  return store;
+  return useContext(ERROR_CONTEXT);
```

Comment:
Can we just return `error` object here?

Reply:
Let me try!
Thanks mate for your suggestion!